src/apm/php.go

```diff
@@ -45,52 +62,57 @@ func InjectPhpagent(phpSpec v1alpha1.Php, pod corev1.Pod, index int) (corev1.Pod
 		}
 	}
 
-	const (
-		phpConcatEnvValues = false
-		concatEnvValues    = true
-	)
-
-	setPhpEnvVar(container, envPhpsymbolicOption, phpSymbolicOptionArgument, phpConcatEnvValues)
-
-	setPhpEnvVar(container, envPhpSilentOption, phpSilentOptionArgument, phpConcatEnvValues)
-
-	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+	// define instrumentation artifact volume
+	instVolume := corev1.VolumeMount{
 		Name:      volumeName,
 		MountPath: "/newrelic-instrumentation",
-	})
+	}
+
+	setPhpEnvVar(container, envIniScanDirKey, envIniScanDirVal, true)
```

Comment:
Can you expand on that a little more? I don't follow why this would prohibit custom INI search paths. If you look at the logic of the function, we aren't overwriting this environment variable, we're concatenating/appending (the `true` arg value) the path that we've defined to the existing definition (if one exists)

Reply:
Looks like if the concatenation occurs the env var will now have a "::" where the two strings meet.  I'm not sure if this is valid or not so might be good to check.
It is a bit contrived but if the customer did not want to default INI search path included this current implementation would not allow that.

Reply:
The name of the function `setPhpEnvVar` does not indicate that it actually either sets or updates `PHP_INI_SCAN_DIR`.

Reply:
> The name of the function `setPhpEnvVar` does not indicate that it actually either sets or updates `PHP_INI_SCAN_DIR`.

This function is a generic implementation part of the original baseline for setting container environment variables that I chose to repurpose from injecting the `NR_INSTALL_USE_CP_NOT_LN` and `NR_INSTALL_SILENT` envvars. We can change it to something more specific for `PHP_INI_SCAN_DIR`, but if we end up having to inject any other variables in the future, we'll have to change it back again.

Reply:
> Looks like if the concatenation occurs the env var will now have a "::" where the two strings meet. I'm not sure if this is valid or not so might be good to check.

I'll fix this to only use one `:`.

> It is a bit contrived but if the customer did not want the default INI search path included this current implementation would not allow that.

This is required for this to work for the PHP agent. Any user who doesn't want to use this will be unable to use the agent operator for php.

Reply:
removing the extra `:` character: 89092db57c7b829e498486104c626668aa6794a6

Reply:
My understanding is the initial `:` is necessary so the default PHP ini path is maintained, otherwise I think everything will just stop working for the PHP environment.
My comment was that when appending is there any special care required to combine the end of an existing `PHP_INI_SCAN_DIR` with the value the PHP agent needs.

Reply:
The `:` is already being prepended in the `setPhpEnvVar` function [here](https://github.com/newrelic/k8s-agents-operator/pull/79/files#diff-9d8393bab101cb85655446396e995080f844370d607c6b39ae4e560f684e138aR128), so the one hardcoded in the string was unnecessary.

Reply:
> My comment was that when appending is there any special care required to combine the end of an existing PHP_INI_SCAN_DIR with the value the PHP agent needs.

This is all being treated like a normal unix environment variable concatenation much like you'd do for your `PATH` variable.

Reply:
I think the sequence `::` is necessary in the ini path somewhere as that is what triggers PHP looking at the default `conf.d` location which all extension and other INI files are usually stored.   Without this I think it will break PHP environments.

Reply:
Met with Brian and we discussed this - I believe the current form is correct.
- If the customer set `PHP_INI_SCAN_DIR` we want to just append our directory to the list and not include the default search path.  The customer will have included it in their value if they wanted it searched, and we should not add it.
- If the customer did not set `PHP_INI_SCAN_DIR` then this works because the path will be set to a `:` followed by the location of the INI file for the agent.  The leading `:` will cause the default search path to be included.  If the customer left the value undefined it means PHP would have searched the default path anyways so we are replicating the desired behavior.